driver-core/src/main/com/mongodb/client/result/bulk/ClientBulkWriteResult.java

```diff
@@ -0,0 +1,138 @@
+ * limitations under the License.
+ */
+package com.mongodb.client.result.bulk;
+
+import com.mongodb.ClientBulkWriteException;
+import com.mongodb.WriteConcern;
+import com.mongodb.annotations.Evolving;
+import com.mongodb.bulk.WriteConcernError;
+import com.mongodb.client.model.bulk.ClientBulkWriteOptions;
+import com.mongodb.client.model.bulk.ClientWriteModel;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * The result of a successful or partially successful client-level bulk write operation.
+ * Note that if only some of the {@linkplain ClientWriteModel individual write operations} succeed,
+ * or if there are {@link WriteConcernError}s, then the successful partial result
+ * is still accessible via {@link ClientBulkWriteException#getPartialResult()}.
+ *
+ * @see ClientBulkWriteException
+ * @since 5.3
+ */
+@Evolving
+public interface ClientBulkWriteResult {
+    /**
+     * Indicated whether this result was {@linkplain WriteConcern#isAcknowledged() acknowledged}.
+     * If not, then all other methods throw {@link UnsupportedOperationException}.
+     *
+     * @return Whether this result was acknowledged.
+     */
+    boolean isAcknowledged();
+
+    /**
+     * The number of documents that were inserted across all insert operations.
+     *
+     * @return The number of documents that were inserted.
+     * @throws UnsupportedOperationException If this result is not {@linkplain #isAcknowledged() acknowledged}.
+     */
+    long getInsertedCount();
+
+    /**
+     * The number of documents that were upserted across all update and replace operations.
+     *
+     * @return The number of documents that were upserted.
+     * @throws UnsupportedOperationException If this result is not {@linkplain #isAcknowledged() acknowledged}.
+     */
+    long getUpsertedCount();
+
+    /**
+     * The number of documents that matched the filters across all operations with filters.
+     *
+     * @return The number of documents that were matched.
+     * @throws UnsupportedOperationException If this result is not {@linkplain #isAcknowledged() acknowledged}.
+     */
+    long getMatchedCount();
+
+    /**
+     * The number of documents that were modified across all update and replace operations.
+     *
+     * @return The number of documents that were modified.
+     * @throws UnsupportedOperationException If this result is not {@linkplain #isAcknowledged() acknowledged}.
+     */
+    long getModifiedCount();
+
+    /**
+     * The number of documents that were deleted across all delete operations.
+     *
+     * @return The number of documents that were deleted.
+     * @throws UnsupportedOperationException If this result is not {@linkplain #isAcknowledged() acknowledged}.
+     */
+    long getDeletedCount();
+
+    /**
+     * The {@linkplain ClientBulkWriteOptions#verboseResults(Boolean) verbose results} of individual operations.
+     *
+     * @return {@link Optional} verbose results of individual operations.
```

Comment:
Suggest: Adding `@see ClientBulkWriteOptions#verboseResults(Boolean)` as it explicitly links to the option and is more obvious in the javadoc.

Reply:
Done in a11e5f6e46970321b86bc5e23b5762e2838ec5e5.